include/oneapi/dpl/internal/iterator_impl.h

```diff
@@ -45,7 +45,7 @@ struct extract_accessor<oneapi::dpl::__internal::sycl_iterator<Mode, T, Allocato
     static constexpr sycl::access::mode mode = Mode;
     static constexpr int dim = 1;
     using buffer_type = sycl::buffer<T, dim, Allocator>;
-    using accessor_type = sycl::accessor<T, dim, mode, sycl::access::target::host_buffer>;
+    using accessor_type = sycl::accessor<T, dim, mode, sycl::access::target::host_task>;
```

Comment:
I think we should instead use `host_accessor` here. @MikeDvorskiy please take a look.

Reply:
I agree that `sycl::host_accessor` should be used as the return type.

Also, the sycl buffer member function used to get the return value,  `get_access<mode>()`, is also deprecated in SYCL 2020 and `get_host_access()` should be used instead.  These should change together.